projects/dex-ui/src/components/Swap/TokenInput.tsx

```diff
@@ -139,7 +139,7 @@ export const TokenInput: FC<TokenInput> = ({
 
       {showBalance && (
         <BalanceRow>
-          <Balance>Balance {isBalanceLoading ? <Spinner size={12} /> : balance?.[token.symbol].toHuman()}</Balance>
+          <Balance onClick={balance ? () => updateAmount(balance?.[token.symbol].toHuman()) : undefined}>Balance: {isBalanceLoading ? <Spinner size={12} /> : balance?.[token.symbol].toHuman()}</Balance>
```

Comment:
There's already a function for this, `handleClickMax()`, we used to have a "Max" button. Just assign that method here